client/src/state/schema/SchemaReducer.js

```diff
@@ -12,6 +12,18 @@ export default function schemaReducer(state = initialState, action = null) {
       }));
     }
 
+    case ACTION_TYPES.CLEAR_SCHEMA: {
+      return deepFreeze(Object.entries(state).reduce((prev, [key, value]) => {
```

Comment:
fair enough, I can move this to a helper function... this is basically removing an object property the same as `const newState = { ...state }; delete state[action.payload.id];` but without creating the `newState` object.